src/main/java/beam/analysis/plots/GraphUtils.java

```diff
@@ -160,6 +171,23 @@ public static void plotLegendItems(CategoryPlot plot, List<String> legendItemNam
         plot.setFixedLegendItems(legendItems);
     }
 
+    public static void plotLegendItemsWithColors(CategoryPlot plot, List<String> legendItemNames, Map<String, Color> colorMap) {
+        LegendItemCollection legendItems = new LegendItemCollection();
+        for (int i = 0; i < legendItemNames.size(); i++) {
+            String legendName = legendItemNames.get(i);
+            Color color;
+            if (colorMap.containsKey(legendName)) {
+                color = colorMap.get(legendName);
+            } else {
+                log.error(String.format("There is no color configured for [%s]", legendItemNames.get(i)));
```

Comment:
Suggestion: Here `log.error` is more restrictive. make it either `info` or `warn` because eventually if color not defined then we are generating new color

Reply:
Done. I would also like to make the graphs legend consistent, so the car types always go in alphabetical order.